### PR TITLE
3 new APIs added to SignalR service and rename the sample files

### PR DESCRIPTION
**Pull request by juniwang**
### Latest improvements:
- We add a new API to `Restart` to restart a SignalR resource
The APIs are already available in RP and can be invoked on azure portal. Now we need to enable customer to do it via SDKs/cli/powershell.
- We also renamed existing sample files. Since we find the former names are easy to conflict. Now we use `OperationId` which is unique through SignalR service.
- Suppress several warnings. See the details in the md file.

### Tests performed
Tests documented at https://github.com/Azure/adx-documentation-pr/wiki/Azure-Swagger-Tools passed.

### CODE REVIEW

**Comment by juniwang** [comment]
Can one of the ARM admins verify this patch? thanks a lot

**Comment by juniwang** [comment]
@KrisBash @ravbhatnagar Hi, could you please review the PR? thanks

File: specification/signalr/resource-manager/Microsoft.SignalRService/stable/2018-10-01/signalr.json
Comment (ravbhatnagar) [inline specification/signalr/resource-manager/Microsoft.SignalRService/stable/2018-10-01/signalr.json]: camelcase resource type names. so signalR

File: specification/signalr/resource-manager/Microsoft.SignalRService/stable/2018-10-01/signalr.json
Comment (ravbhatnagar) [inline specification/signalr/resource-manager/Microsoft.SignalRService/stable/2018-10-01/signalr.json]: model the final response as well. It would help the client getting the final response. So if the final response is 200, please add it to responses. And add the schema of the response if there is one.

File: specification/signalr/resource-manager/Microsoft.SignalRService/stable/2018-10-01/signalr.json
Comment (ravbhatnagar) [inline specification/signalr/resource-manager/Microsoft.SignalRService/stable/2018-10-01/signalr.json]: add a "default" response for all APIs. Check other RPs as example.

File: specification/signalr/resource-manager/Microsoft.SignalRService/stable/2018-10-01/signalr.json
Comment (ravbhatnagar) [inline specification/signalr/resource-manager/Microsoft.SignalRService/stable/2018-10-01/signalr.json]: make it features instead of listFeatures. Also, I want to understand more about this resource type? What are these features? Can you give an example? From what i think it does, it might make sense to have these "features" as properties on the resource itself which can be enabled/disabled by PUT/PATCH operations on the signalR resource itself.

**Review by ravbhatnagar — CHANGES_REQUESTED** [review]
Lets have a chat about the features/switchFeatures API. I dont think i have full context.

File: specification/signalr/resource-manager/Microsoft.SignalRService/stable/2018-10-01/signalr.json
Comment (juniwang) [reply specification/signalr/resource-manager/Microsoft.SignalRService/stable/2018-10-01/signalr.json]: Sure we can update this. But we already used `SignalR` in all other APIs as well as documents. Can we keep capitalized version here to be consistent?

File: specification/signalr/resource-manager/Microsoft.SignalRService/stable/2018-10-01/signalr.json
Comment (juniwang) [reply specification/signalr/resource-manager/Microsoft.SignalRService/stable/2018-10-01/signalr.json]: Currently 202 is the only response by design. This API will trigger a long-running backend operation which will complete asynchronously. That's why we didn't include 200/201. 200 might give the caller a wrong sense that the operation is completed but actually not.  As for 201, we actually don't create anything in this operation.

We saw the warnings while validating the spec. That's why we suppress it in the readme. Just because 200/201/204 all makes us feel strange

File: specification/signalr/resource-manager/Microsoft.SignalRService/stable/2018-10-01/signalr.json
Comment (juniwang) [reply specification/signalr/resource-manager/Microsoft.SignalRService/stable/2018-10-01/signalr.json]: Basically a SignalR resource has some backend containers running in ACS, where each container is running a SignalR runtime written in C#. Those features are actually configurations of the SignalR runtime. For now, we only have one feature named `ServiceMode`(either Default, Classic or Serverless), which corresponds to different customer scenarios. SignalR runtime will read feature values and behave correspondingly. 

For us, SignalR service management and Service runtime configuration is quite separated. As I said, the runtime is written in C#(and SignalR is essentially a asp.net library) which means we actually have many C#/SignalR configurations that can change the realtime behavior. We may gradually open some settings to customers. That's why we may want a new resource type dedicated for these kind of things, for better separation and flexibility

**Comment by juniwang** [comment]
@ravbhatnagar Please review again

File: specification/signalr/resource-manager/Microsoft.SignalRService/stable/2018-10-01/signalr.json
Comment (juniwang) [reply specification/signalr/resource-manager/Microsoft.SignalRService/stable/2018-10-01/signalr.json]: After some internal discussion, we decided to deprecate those two APIs: `ListFeature` and `SwitchFeature`. We will use propertes instead, just like you said. So I removed those two spec from swagger file

File: specification/signalr/resource-manager/Microsoft.SignalRService/stable/2018-10-01/signalr.json
Comment (juniwang) [reply specification/signalr/resource-manager/Microsoft.SignalRService/stable/2018-10-01/signalr.json]: added

File: specification/signalr/resource-manager/Microsoft.SignalRService/stable/2018-10-01/signalr.json
Comment (juniwang) [reply specification/signalr/resource-manager/Microsoft.SignalRService/stable/2018-10-01/signalr.json]: added 204

File: specification/signalr/resource-manager/Microsoft.SignalRService/stable/2018-10-01/signalr.json
Comment (juniwang) [reply specification/signalr/resource-manager/Microsoft.SignalRService/stable/2018-10-01/signalr.json]: Updated

**Review by KrisBash — APPROVED** [review]
ARM changes appear to be addressed. Thanks!

**Comment by juniwang** [comment]
Thanks Kris. @ravbhatnagar Could you please review again?

**Comment by juniwang** [comment]
Can one of the admin merge this? We are approaching our SLA to expose the new API. Thanks a lot.

**Comment by adxsdknet** [comment]
# Automation for azure-sdk-for-net 
A PR has been created for you: 
https://github.com/azure/azure-sdk-for-net/pull/5505 
.NET SDK Commits: 
https://github.com/adxsdknet/azure-sdk-for-net/pull/5505/commits/c28dbf52198969ab9c6abfa69a7043ceb5e8324b

**Review by anuchandy — APPROVED** [review]

**Comment by anuchandy** [comment]
@juniwang just requested @ravbhatnagar's sign-off, which is required to merge this.

**Comment by ravbhatnagar** [comment]
Signing off from ARM side.

**Comment by anuchandy** [comment]
@ravbhatnagar Coud you go to "File Changes -> Review Changes -> Approve" then approve it please? since you requested changes in the past, it won't let me merge without your explicit approval (merge button is disabled for me)

**Review by ravbhatnagar — APPROVED** [review]

**Comment by juniwang** [comment]
Thanks a lot, guys

**Comment by jhendrixMSFT** [comment]
@AutorestCI regenerate azure-sdk-for-go

**Pull request merged**
